ratel-codegen/src/statement.rs

```diff
@@ -309,6 +310,72 @@ impl<'ast, G: Generator> ToCode<G> for SwitchCase<'ast> {
     }
 }
 
+impl<'ast, G: Generator> ToCode<G> for ImportDeclaration<'ast> {
+    #[inline]
+    fn to_code(&self, gen: &mut G) {
+        let mut tail = false;
+        gen.write_bytes(b"import ");
+        if self.specifiers.is_empty() {
+            gen.write_byte(b'\'');
+            gen.write_bytes(self.source.as_bytes());
+            gen.write_byte(b'\'');
+            return
+        }
+
+        for spec in self.specifiers {
```

Comment:
There is probably a way to make this more DRY.

Could try using `gen.write_list` here with a `filter` or `filter_map` iterators if the inner types got `ToCode` implemented for them. This would also avoid having to keep track of the `tail` boolean, which seems like an easy thing to get wrong if this ever gets refactored.

Reply:
`gen.write_list` does not seem to accept `Vec<ForImportSpecifier>`.
How can I implement it?

Reply:
`write_list` operates on `IntoInterator` types, so you can skip collecting to `Vec` part:

```rust
let specs = self.specifiers.iter().filter_map(|spec| {
    match spec.item {
        ForImportSpecifier::ImportDefaultSpecifier(_) |
        ForImportSpecifier::ImportNamespaceSpecifier(_) => Some(spec.item),
        _ => None
    }
});
gen.write_list(specs);
```